types/wlr_idle_inhibit_v1.c

```diff
@@ -26,9 +26,11 @@ wlr_idle_inhibitor_v1_from_resource(struct wl_resource *resource) {
 	return wl_resource_get_user_data(resource);
 }
 
-static void idle_inhibitor_destroy(struct wl_resource *resource) {
-	struct wlr_idle_inhibitor_v1 *inhibitor =
-		wlr_idle_inhibitor_v1_from_resource(resource);
+static void idle_inhibitor_v1_destroy(struct wlr_idle_inhibitor_v1 *inhibitor) {
+	if (!inhibitor) {
+		return;
+	}
+	wl_resource_set_user_data(inhibitor->resource, NULL);
```

Comment:
We should probably do this after emitting the event. Some listener may be calling `wlr_idle_inhibitor_v1_from_resource`.

Reply:
Right, force-pushed. Head is turned off at this point so this comes straight from CONTRIBUTING.md, if/when you update it since the original `subsurface_destroy` has an emit event as well I'd suggest putting it in the "template" as well